modules/core/src/main/scala/eu/timepit/scalasteward/application/Cli.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 scala-steward contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.timepit.scalasteward.application
+
+import caseapp._
+import cats.effect.Sync
+import cats.implicits._
+
+trait Cli[F[_]] {
+  def parseArgs(args: List[String]): F[Cli.Args]
+}
+
+object Cli {
+
+  final case class Args(
+      workspace: String,
+      reposDir: Option[String],
+      reposFile: String,
+      gitAuthorName: String,
+      gitAuthorEmail: String,
+      githubApiHost: String,
+      githubLogin: String,
+      gitAskPass: String
+  )
+
+  def create[F[_]](implicit F: Sync[F]): Cli[F] = new Cli[F] {
```

Comment:
It seems to me that `ApplicativeError[F, Throwable]` would be enough here. There is also a handy `ApplicativeThrowable` for the former in the `util` package object

Reply:
Good point!